fpdf/fpdf.py

```diff
@@ -366,6 +366,10 @@ def __init__(
         self._drawing_graphics_state_registry = GraphicsStateDictRegistry()
         # map page numbers to a set of GraphicsState names:
         self.graphics_style_names_per_page_number = defaultdict(set)
+        self.shadings_per_page_number = defaultdict(set)
+        self.patterns_per_page_number = defaultdict(set)
+        self._shading_registry = {}
+        self._pattern_registry = {}
```

Comment:
Would it be relevant to group those attributes into a single object, in order to limit the number of attributes added to `FPDF` instances? (in order to improve code readability)